tests/responses/test_fileresponse.py

```diff
@@ -11,25 +12,135 @@
 
 class FileResponseTests(SimpleTestCase):
     def test_file_from_disk_response(self):
-        response = FileResponse(open(__file__, 'rb'))
+        filenames = (
+            ('', 'test_fileresponse.py'),
+            ('custom_name.py', 'custom_name.py'),
+        )
+        dispositions = (
+            (False, 'inline'),
+            (True, 'attachment'),
+        )
+        for (filename, header_filename), (as_attachment, header_disposition)\
+                in itertools.product(filenames, dispositions):
+            with self.subTest(filename=filename, disposition=header_disposition):
+                response = FileResponse(open(__file__, 'rb'), filename=filename, as_attachment=as_attachment)
+                self.assertEqual(response['Content-Length'], str(os.path.getsize(__file__)))
+                self.assertIn(response['Content-Type'], ['text/x-python', 'text/plain'])
+                self.assertEqual(response['Content-Disposition'],
+                                 '%s; filename="%s"' % (header_disposition, header_filename))
+                response.close()
+
+    def test_file_from_buffer_response(self):
+        filenames = (
+            ('', 'test_fileresponse.py'),
+            ('custom_name.py', 'custom_name.py'),
+        )
+        dispositions = (
+            (False, 'inline'),
+            (True, 'attachment'),
+        )
+        content_types = (
+            'text/plain',
+            'text/html',
+        )
+        for (filename, header_filename), (as_attachment, header_disposition), content_type\
+                in itertools.product(filenames, dispositions, content_types):
+            with self.subTest(filename=header_filename, disposition=header_disposition, content_type=content_type):
+                response = FileResponse(io.BytesIO(b'binary content'),
+                                        filename=filename,
+                                        as_attachment=as_attachment,
+                                        content_type=content_type)
+                self.assertEqual(response['Content-Length'], '14')
+                self.assertEqual(response['Content-Type'], content_type)
```

Comment:
Are we really testing the Content-Type setting here? 

It strikes me this test is really about the Content-Disposition behaviour. We should focus it to that. (With a separate case for testing 'Content-Type' branched (such as guessing the content type from the filename.)